lib/src/default_index/revset_engine.rs

```diff
@@ -709,6 +709,96 @@ pub fn evaluate<I: AsCompositeIndex + Clone>(
     Ok(RevsetImpl::new(internal_revset, index))
 }
 
+#[derive(Clone, Eq, PartialEq)]
```

Comment:
nit: `Copy` could be added since the implementation appears to assume that `UnionFindNode` can be cheaply cloned